### PR TITLE
Fxing the paths that are resolved relative to a network share

### PR DESCRIPTION
**Pull request by ramarag**
@JohnChen0 @JeremyKuhne  PTAL

### CODE REVIEW

File: src/utilcode/longfilepathwrappers.cpp [hunk 1]
```diff
@@ -1366,12 +1366,26 @@ HRESULT LongFile::NormalizePath(SString & path)
...
 
-
-    //wcscpy_s always termintes with NULL, so we are saving the character that will be overwriiten
-    WCHAR temp = buffer[prefix.GetCount()];
-    wcscpy_s(buffer, prefix.GetCount() + 1, prefix.GetUnicode());
-    buffer[prefix.GetCount()] = temp;
-    path.CloseBuffer(ret + prefixLen);
+	SString fullpath(SString::Literal, buffer + prefixLen);
+
+	if (fullpath.BeginsWith(UNCPathPrefix) && prefixLen != prefix.GetCount() - 2)
```
Comment (JeremyKuhne) [inline src/utilcode/longfilepathwrappers.cpp]: Can you just check prefix.BeginsWith(UNCExtendedPathPrefix)?

Comment (ramarag) [inline src/utilcode/longfilepathwrappers.cpp]: This is the case where the normalization of a relative path returns a UNC path. if we knew this was gonna happen then we would take the else path

File: src/utilcode/longfilepathwrappers.cpp [hunk 2]
```diff
@@ -1366,12 +1366,26 @@ HRESULT LongFile::NormalizePath(SString & path)
...
+	{
+		// must initialize before calling SString::Empty()
+		SString::Startup();
```
Comment (JohnChen0) [inline src/utilcode/longfilepathwrappers.cpp]: Can we actually get here without already having called `SString::Startup()`? Also, I don't see `SString::Empty()` being called below. Is it called indirectly by another method? Might be a good idea to clarity in the comment.

File: src/utilcode/longfilepathwrappers.cpp [hunk 3]
```diff
@@ -1366,12 +1366,26 @@ HRESULT LongFile::NormalizePath(SString & path)
...
+
+		fullpath.Delete(fullpath.Begin(), 2);
```
Comment (JohnChen0) [inline src/utilcode/longfilepathwrappers.cpp]: Suggest comment about why deleting 2 characters.

**Comment by JohnChen0** [comment]
LGTM

File: src/utilcode/longfilepathwrappers.cpp (hunk 2 above)
Comment (ramarag) [inline src/utilcode/longfilepathwrappers.cpp]: done

File: src/utilcode/longfilepathwrappers.cpp (hunk 3 above)
Comment (ramarag) [inline src/utilcode/longfilepathwrappers.cpp]: done

File: src/utilcode/longfilepathwrappers.cpp [hunk 4]
```diff
@@ -1366,12 +1367,28 @@ HRESULT LongFile::NormalizePath(SString & path)
         }
     }
 
-
-    //wcscpy_s always termintes with NULL, so we are saving the character that will be overwriiten
-    WCHAR temp = buffer[prefix.GetCount()];
-    wcscpy_s(buffer, prefix.GetCount() + 1, prefix.GetUnicode());
-    buffer[prefix.GetCount()] = temp;
-    path.CloseBuffer(ret + prefixLen);
+	SString fullpath(SString::Literal, buffer + prefixLen);
```
Comment (JohnChen0) [inline src/utilcode/longfilepathwrappers.cpp]: These lines are indented by tabs. Please change them to spaces.

File: src/utilcode/longfilepathwrappers.cpp [hunk 5]
```diff
@@ -1366,12 +1367,28 @@ HRESULT LongFile::NormalizePath(SString & path)
...
+
+	//Check if the resolved path is a UNC. By default we assume relative path to resolve to disk 
+	if (fullpath.BeginsWith(UNCPathPrefix) && prefixLen != prefix.GetCount() - 2)
+	{
+		// must initialize before calling SString::Empty() via SString::Delete
+		SString::Startup();
+
+		//Remove the leading '\\' from the UNC path to be replaced with UNCExtendedPathPrefix
+		fullpath.Delete(fullpath.Begin(), 2);
+		fullpath.Insert(fullpath.Begin(), UNCExtendedPathPrefix);
```
Comment (JohnChen0) [inline src/utilcode/longfilepathwrappers.cpp]: More efficient to combine `fullpath.Delete` and `fullpath.Insert` with `fullpath.Replace`.

File: src/utilcode/longfilepathwrappers.cpp (hunk 4 above)
Comment (ramarag) [inline src/utilcode/longfilepathwrappers.cpp]: done

File: src/utilcode/longfilepathwrappers.cpp (hunk 5 above)
Comment (ramarag) [inline src/utilcode/longfilepathwrappers.cpp]: done

File: src/utilcode/longfilepathwrappers.cpp (hunk 1 above)
Comment (JeremyKuhne) [inline src/utilcode/longfilepathwrappers.cpp]: How does this even happen? Can you give an example? You can't set a working directory to a UNC.

Comment (JohnChen0) [inline src/utilcode/longfilepathwrappers.cpp]: PowerShell allows you to set working directory to a UNC path (simply do `cd \\foo\bar`), even though cmd.exe doesn't allow it.

**Comment by ramarag** [comment]
@dotnet-bot test OSX x64 Checked Build and Test 
Ubuntu x64 Checked Build and Test

**Pull request merged**